### ISSUE TITLE
default values for text_field / label_field for datasets

### ISSUE THREAD

**Issue by dhpollack** [pytorch/text#117]
I was just curious if there is a reason the defaults for text_field and label_field are not data.Field() and data.Field(sequential=False).  It's a little confusing from the documentation what these fields should be.  For example, with the imdb sentiment dataset, the example in test has two examples that don't seem consistent.  But if we added default values to the text_field and label_field arguments, one could do something like so:

    class IMDB(data.Dataset):
        ...
        def __init__(self, path, text_field=data.Field(), label_field=data.Field(sequential=False), ... )

then

Example 1:  

    TEXT = data.Field(lower=True, include_lengths=True, batch_first=True)
    LABEL = data.Field(sequential=False)
    imdb = datasets.IMDB("data", text_field=TEXT, label_field=LABEL)
    train, test = datasets.IMDB.splits(TEXT, LABEL)
    ...

Example 2:  

    imdb = datasets.IMDB("data")
    train_iter, test_iter = imdb.iters(batch_size=4)

But perhaps there is a reason why these don't have default values.

**Comment by jekbradbury** [comment]
There would need to be an easy way to access the created Fields after the dataset is built (you can access them as fairly deeply buried attributes, I think, but I haven't seen anyone do it). We do occasionally provide `.iters` for automatic/default dataset creation that creates its own fields.
